core/generator.ts

```diff
@@ -441,6 +437,10 @@ export class CodeGenerator {
     return msg.replace(/%1/g, "'" + id + "'");
   }
 
+  abstract quote_(toQuote: string): string;
```

Comment:
If we can't make this `abstract`, could we instead a default implementation that will work adequately in most languages?  I note that at the moment `JavascriptGenerator`, `LuaGenerator` and `PhpGenerator` all use the same implementation (though it is wrong for at least JavaScript 🙁).